include/deal.II/matrix_free/face_setup_internal.h

```diff
@@ -1041,6 +1042,20 @@ namespace internal
         }
       else
         info.face_orientation = exterior_face_orientation;
+
+      // make sure to select correct subface index in case of non-standard
+      // orientation of the coarser neighbor face
+      if (cell->level() > neighbor->level() && exterior_face_orientation > 0)
+        {
+          Table<2, unsigned int> orientation =
+            ShapeInfo<double>::compute_orientation_table(2);
+          std::array<unsigned int, 8> inverted_orientations{
+            {0, 1, 2, 3, 6, 5, 4, 7}};
```

Comment:
I am wondering if we should have this at a central place?

Reply:
I think so, but I won't have time today and tomorrow to fix it. Maybe we can do that in a follow-up PR?